src/testdrive/src/action/kafka/create_topic.rs

```diff
@@ -170,6 +173,19 @@ impl Action for CreateTopicAction {
         // breaches the default 7-day retention policy.
         .set("retention.ms", "-1")
         .set("compression.type", &self.compression);
+
+        // agressive compaction, when it is enabled
+        let new_topic = if self.compaction {
+            new_topic
+                .set("cleanup.policy", "compact")
```

Comment:
I forget how setting `retention.ms` interacts with setting a non time based `cleanup.policy` -- im assuming cleanup policy is preferred?

Reply:
As I understood it from the docs, `retention.ms` is ignored when `cleanup.policy=compact`: https://kafka.apache.org/documentation/#topicconfigs_retention.ms.

By the way, you can also set `cleanup.policy=compact,delete`: https://cwiki.apache.org/confluence/display/KAFKA/KIP-71%3A+Enable+log+compaction+and+deletion+to+co-exist